truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/nodes/Node.java

```diff
@@ -119,14 +127,31 @@ public NodeCost getCost() {
     }
 
     /**
-     * Clears any previously assigned guest language source code from this node.
+     * @deprecated if your node provides source section, override {@link #getSourceSection()} to
+     *             return it - this method will be removed
+     * 
+     *             Clears any previously assigned guest language source code from this node.
      */
+    @Deprecated
     public void clearSourceSection() {
         this.sourceSection = null;
     }
 
     /**
-     * Retrieves the segment of guest language source code that is represented by this Node.
+     * Retrieves the segment of guest language source code that is represented by this Node. The
+     * default implementation of this method returns <code>null</code>. If your node represents a
+     * segment of the source code, override this method and return a <code>final</code> or
+     * {@link CompilationFinal} field in your node to the caller.
+     *
+     * To define node with <em>fixed</em> {@link SourceSection} that doesn't change after node
+     * construction use:
+     *
+     * {@codesnippet source.section.fixed}
+     *
+     * To create a node which can associate and change the {@link SourceSection} later at any point
+     * of time use:
+     *
+     * {@codesnippet source.section.mutable}
      *
      * @return the source code represented by this Node
      */
```

Comment:
If you use just non-deprecated API it returns null, so I think updating the Javadoc "in advance" is proper thing to do.

----- Reply message -----
Od: "Stefan Marr" <notifications@github.com>
Komu: "graalvm/truffle" <truffle@noreply.github.com>
Kopie: "Jaroslav Tulach" <jaroslav.tulach@oracle.com>
Předmět: [truffle] Source section less node (#35)
Datum: čt, úno 4, 2016 12:06

In truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/nodes/Node.java:
>      public void clearSourceSection() {
>          this.sourceSection = null;
>      }
>  
>      /**
> -     * Retrieves the segment of guest language source code that is represented by this Node.
> +     * Retrieves the segment of guest language source code that is represented by this Node. The
> +     * default implementation of this method returns <code>null</code>. If your node represents a
> +     * segment of the source code, override this method and return a <code>final</code> or
> +     * {@link CompilationFinal} field in your node to the caller.
> +     *
> +     * To define node with <em>fixed</em> {@link SourceSection} that doesn't change after node
> +     * construction use:
> +     *
> +     * {@codesnippet source.section.fixed}
> +     *
> +     * To create a node which can associate and change the {@link SourceSection} later at any point
> +     * of time use:
> +     *
> +     * {@codesnippet source.section.mutable}
>       *
>       * @return the source code represented by this Node
>       */

Does this comment need a transitory note? Mostly, because the implementation does not yet return null?

&mdash;
Reply to this email directly or view it on GitHub.